targets/TARGET_RENESAS/TARGET_RZ_A1XX/ethernet_api.c

```diff
@@ -315,6 +321,61 @@ void ethernetext_set_link_mode(int32_t link) {
     lan_reg_set(link);          /* E-DMAC, E-MAC initialization */
 }
 
+void ethernetext_add_multicast_group(const uint8_t *addr) {
+    uint32_t cnt;
+    uint32_t tmp_data;
+
+    for (cnt = 0; cnt < 32; cnt++) {
+        if ((tsu_ten_tmp & (1 << cnt)) == 0) {
+            tmp_data = ((uint32_t)addr[0] << 24) | ((uint32_t)addr[1] << 16) | ((uint32_t)addr[2] << 8) | ((uint32_t)addr[3]);
+            ETHER_FROM_TSU_ADRH0_ARRAY[cnt]->TSU_ADRH0 = tmp_data;
+            tmp_data = ((uint32_t)addr[4] << 8) | ((uint32_t)addr[5]);
+            ETHER_FROM_TSU_ADRH0_ARRAY[cnt]->TSU_ADRL0 = tmp_data;
+            while ((ETHERTSU_ADSBSY & 0x0001) != 0) {
+                ;
+            }
+            ETHERTSU_TEN |= (1 << cnt);
+            tsu_ten_tmp  |= (1 << cnt);
+            break;
+        }
+    }
```

Comment:
Unlikely to happen, but if you run out of room in the 32-entry table, you should do "set all multicast". You can't leave the caller unable to receive the group they asked for.